test/e2e/docker_containers.go

```diff
@@ -0,0 +1,148 @@
+		pod := entrypointTestPod()
+		pod.Spec.Containers[0].Args = []string{"/cmd-2", "override-cmd"}
+
+		runEntrypointTest("override cmd", c, api.NamespaceDefault, pod, []string{
+			"[/ep-1 /cmd-2 override-cmd",
+			"[/cmd-2 override-cmd]",
+		})
+	})
+
+	// Note: when you override the entrypoint, the image's cmd is ignored
+	It("should be able to override the entrypoint of the image", func() {
+		pod := entrypointTestPod()
+		pod.Spec.Containers[0].Command = []string{"/ep-2"}
+
+		runEntrypointTest("override entrypoint", c, api.NamespaceDefault, pod, []string{
+			"[/ep-2]",
+		})
+	})
+
+	It("should be able to override the entrypoint and cmd of the image", func() {
+		pod := entrypointTestPod()
+		pod.Spec.Containers[0].Command = []string{"/ep-2"}
+		pod.Spec.Containers[0].Args = []string{"/cmd-2", "override-cmd"}
+
+		runEntrypointTest("override entrypoint", c, api.NamespaceDefault, pod, []string{
+			"[/ep-2 /cmd-2 override-cmd]",
+			"[/cmd-2 override-cmd]",
+		})
+	})
+})
+
+const testContainerName = "test-container"
+
+// Return a prototypical entrypoint test pod
+func entrypointTestPod() *api.Pod {
+	podName := "client-containers-" + string(util.NewUUID())
+
+	return &api.Pod{
+		ObjectMeta: api.ObjectMeta{
+			Name: podName,
+		},
+		Spec: api.PodSpec{
+			Containers: []api.Container{
+				{
+					Name:  testContainerName,
+					Image: "kubernetes/eptest:0.1",
+				},
+			},
+			RestartPolicy: api.RestartPolicyNever,
+		},
+	}
+}
+
+// pod must have a container named 'test-container'
+func runEntrypointTest(scenarioName string, c *client.Client, ns string, pod *api.Pod, expectedOutput []string) {
+	By(fmt.Sprintf("Creating a pod to test %v", scenarioName))
+	// Make a client pod that verifies that it has the service environment variables.
+
+	defer c.Pods(ns).Delete(pod.Name)
+	if _, err := c.Pods(ns).Create(pod); err != nil {
+		Failf("Failed to create pod: %v", err)
+	}
+	// Wait for client pod to complete.
+	expectNoError(waitForPodSuccessInNamespace(c, pod.Name, testContainerName, ns))
+
+	// Grab its logs.  Get host first.
+	podStatus, err := c.Pods(ns).Get(pod.Name)
+	if err != nil {
+		Failf("Failed to get pod to know host: %v", err)
+	}
+	By(fmt.Sprintf("Trying to get logs from host %s pod %s container %s: %v",
+		podStatus.Status.Host, podStatus.Name, podStatus.Spec.Containers[0].Name, err))
+	var logs []byte
+	start := time.Now()
+
+	// Sometimes the actual containers take a second to get started, try to get logs for 60s
+	for time.Now().Sub(start) < (60 * time.Second) {
+		logs, err = c.Get().
+			Prefix("proxy").
+			Resource("minions").
+			Name(podStatus.Status.Host).
+			Suffix("containerLogs", ns, podStatus.Name, podStatus.Spec.Containers[0].Name).
+			Do().
+			Raw()
+		fmt.Sprintf("pod logs:%v\n", string(logs))
+		By(fmt.Sprintf("pod logs:%v\n", string(logs)))
+		if strings.Contains(string(logs), "Internal Error") {
+			By(fmt.Sprintf("Failed to get logs from host %s pod %s container %s: %v",
```

Comment:
@vmarmol ack; I'm going to make a follow up PR to factor out this code since at this point we have three different tests that can all use it; will address then.